### ISSUE TITLE
capabilities.json is empty !

### ISSUE THREAD

**Issue by solub** [python-escpos/python-escpos#556]
Hi,

I just installed the latest version (3.0a10) of python-escpos from git and am having this error showing up at import:

`from escpos.printer import Usb`

```
File "C:\Users\solub\anaconda3\lib\site-packages\escpos\capabilities.py", line 58, in <module>
    PROFILES: Dict[str, Any] = CAPABILITIES["profiles"]
TypeError: string indices must be integers
```

It seems it's attempting to load the `CAPABILITY` dictionary from a JSON file (`capabilities.json`) but that file only contains an unexpected path to another JSON file:

`../../capabilities-data/dist/capabilities.json`

Don't know what to do from here.

Windows 10
Python 3.8

**Comment by patkan** [comment]
Hi, how exactly have you generated your package?

If you pulled the git repository, you have to also get the submodules. (Capabilities data is included as a git submodule and that has to be initialized) Otherwise, it is possible to "sideload" a capabilities file with an environment variable: https://python-escpos.readthedocs.io/en/latest/user/usage.html#advanced-usage-change-capabilities-profile


Apart from that, is this the full stack trace? If not please provide the full stack trace, that would be helpful.

**Comment by solub** [comment]
Apologies for the late reply.

I've cloned the repository and the submodules then installed the whole like this:

```
git clone https://github.com/python-escpos/python-escpos.git
cd python-escpos
git submodule init
git submodule update
python setup.py install
```

However I still get the same error at launch. Here's the full message:

```
Traceback (most recent call last):
  File "C:\Users\solub\Documents\Thonny Scripts\Test.py", line 1, in <module>
    from escpos.printer import Usb
  File "C:\Users\solub\anaconda3\lib\site-packages\python_escpos-3.0a10.dev27+g2421775-py3.8.egg\escpos\printer\__init__.py", line 4, in <module>
    from .cups import CupsPrinter
  File "C:\Users\solub\anaconda3\lib\site-packages\python_escpos-3.0a10.dev27+g2421775-py3.8.egg\escpos\printer\cups.py", line 14, in <module>
    from ..escpos import Escpos
  File "C:\Users\solub\anaconda3\lib\site-packages\python_escpos-3.0a10.dev27+g2421775-py3.8.egg\escpos\escpos.py", line 24, in <module>
    from escpos.capabilities import get_profile
  File "C:\Users\solub\anaconda3\lib\site-packages\python_escpos-3.0a10.dev27+g2421775-py3.8.egg\escpos\capabilities.py", line 177, in <module>
    class Profile(get_profile_class("default")):
  File "C:\Users\solub\anaconda3\lib\site-packages\python_escpos-3.0a10.dev27+g2421775-py3.8.egg\escpos\capabilities.py", line 158, in get_profile_class
    profiles: Dict[str, Any] = CAPABILITIES["profiles"]
TypeError: string indices must be integers
```

**Comment by patkan** [comment]
Thank you for the additional clarification.

The profile name should be OK. (You can check here: https://python-escpos.readthedocs.io/en/latest/printer_profiles/available-profiles.html#tm-t20ii)


Since you also tried installing from PIP, I guess this is not a issue with the setup.py script on Windows, but to narrow further down, this is the current master as a wheel: [python-escpos-wheel.zip](https://github.com/python-escpos/python-escpos/files/12546553/python-escpos-wheel.zip) You could try installing that.

You could additionally search for a capabilities.json in your site-packages folder (you can see the path in the traceback). It should contain a json with the profiles. Either this file is empty/does not exist or there is an issue with the resource discovery mechanism (I guess under Windows, because I cannot see it locally or in CI)

**Comment by patkan** [comment]
If have had a deeper look at the issue. If you have tried the version 3.0a8, then it is to be expected that the profile TM-T20II is not included as this came with 3.0a9. So with the wheel from my post from before, you should be able to use the profile.

Please try building the wheel first and installing that then. (setup.py bdist_wheel and then pip install the wheel from dist/)

**Comment by solub** [comment]
PIP install of version 3.0a9 did indeed solve the issue, thank you so much!
